src/proxy/outbound.rs

```diff
@@ -201,6 +201,7 @@ impl OutboundConnection {
                     .method(hyper::Method::CONNECT)
                     .version(hyper::Version::HTTP_2)
                     .header("baggage", baggage(&req))
+                    .header("X-Forwarded-For", remote_addr.to_string())
```

Comment:
I think we need to use `Forwarded` header here, since presumably we want to bind to the source IP and port.